src/Symfony/Component/DependencyInjection/Attribute/Parameter.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Symfony\Component\DependencyInjection\Attribute;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER | \Attribute::TARGET_PROPERTY)]
```

Comment:
```suggestion
#[\Attribute(\Attribute::TARGET_PARAMETER)]
```

Reply:
We do allow property injection in public properties in the component though.

Reply:
If we want this to work with public properties, then sure, let's keep but `TaggedIterator`, for instance, does not support this.

Reply:
@kbond That was indeed the intention.